src/OpenTelemetry.AutoInstrumentation.Native/member_resolver.cpp

```diff
@@ -0,0 +1,44 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#include "member_resolver.h"
+
+namespace trace
+{
+
+MemberResolver::MemberResolver(const ComPtr<IMetaDataImport2>& import, const ComPtr<IMetaDataEmit2>& emit)
+    : metadaImport_(import), metadaEmit_(emit)
+{
+}
+
+HRESULT MemberResolver::GetTypeRefOrDefByName(mdToken tkResolutionScope, LPCWSTR szName, mdToken* token)
+{
+    if (TypeFromToken(tkResolutionScope) == mdtAssembly)
+    {
+        tkResolutionScope = mdTokenNil;
+    }
+
+    if (tkResolutionScope == mdTokenNil || TypeFromToken(tkResolutionScope) == mdtTypeDef)
+    {
+        return metadaImport_->FindTypeDefByName(szName, tkResolutionScope, token);
+    }
+
+    // TODO: Prior to metadaEmit_->DefineTypeRefByName it may be good to try
```

Comment:
Are these TODOs something that you plan to follow up on in a separate PR?

Reply:
I will remove them. Per my test CLR runtume does it for us already.